### PR TITLE
Endpoint to fetch user tilesets

### PR DESCRIPTION
**Pull request by dgaubert**

### CODE REVIEW

**Review by jgoizueta — CHANGES_REQUESTED** [review]
Could you check my question about the LIMIT?
Apart from that nice BQ scripting and pagination handling!
We'll have to write tests for this and the connections API at some point but let's go ahead without them by now.

File: app/services/carto/bigquery_tilesets_service.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          ELSE
+            SET metadata_query = CONCAT(metadata_query, ' UNION ALL ', query);
+          END IF;
+        END LOOP;
+
+        EXECUTE IMMEDIATE """
+          SELECT *
+          FROM (""" || metadata_query || """)
+          ORDER BY #{pagination[:order]} #{pagination[:direction]}
+          LIMIT #{pagination[:page] * pagination[:per_page]} OFFSET #{pagination[:offset]}
```
[the remark this answers is not in the document]
Comment (dgaubert) [reply app/services/carto/bigquery_tilesets_service.rb:106]: Oops! good catch!

[the remark this answers is not in the document]
Comment (dgaubert) [reply app/services/carto/bigquery_tilesets_service.rb:106]: Offset is already calculated in https://github.com/CartoDB/cartodb/blob/3ce5c1be50c50e0d134b60c13981848f66a96ea6/app/controllers/carto/api/public/bigquery_tilesets_controller.rb#L68

**Review by jgoizueta — APPROVED** [review]
🏁

**Pull request merged**